internal/types/compare_test.go

```diff
@@ -37,18 +37,6 @@ func TestCompare(t *testing.T) {
 			b:        must.NotFail(NewArray(NullType{})),
 			expected: Less,
 		},
```

Comment:
Removed because they are not the same for comparison, but for sorting they are considered that same. And this test is for comparison. See https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#non-existent-fields